packages/rum-core/test/common/context.spec.js

```diff
@@ -0,0 +1,207 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { addSpanContext, addTransactionContext } from '../../src/common/context'
+import resourceEntries from '../fixtures/resource-entries'
+import Span from '../../src/performance-monitoring/span'
+import Transaction from '../../src/performance-monitoring/transaction'
+import { PAGE_LOAD } from '../../src/common/constants'
+import { mockGetEntriesByType } from '../utils/globals-mock'
+
+describe('Context', () => {
+  function createSpanWithData(url, isXHR = true) {
+    const span = new Span(`GET ${url}`, 'external')
+    const data = {
+      url,
+      method: 'GET'
+    }
+    /**
+     * Testing for both XHR and Fetch spans
+     */
+    if (isXHR) {
+      data.target = {
+        status: 200
+      }
+    } else {
+      data.response = {
+        status: 200
+      }
+    }
+    span.end()
+    addSpanContext(span, data)
+    return span
+  }
+
+  it('should add external span context with all fields', () => {
+    let url = 'http://user:pass@testing.local:1234/path?query'
+    let span = createSpanWithData(url)
+    expect(span.context).toEqual({
+      http: {
+        method: 'GET',
+        url: 'http://[REDACTED]:[REDACTED]@testing.local:1234/path?query',
+        status_code: 200
+      },
+      destination: {
+        service: {
+          name: 'http://testing.local:1234',
+          resource: 'testing.local:1234',
+          type: 'external'
+        },
+        address: 'testing.local',
+        port: '1234'
+      }
+    })
+
+    url = 'https://www.elastic.co:443/products/apm'
+    span = createSpanWithData(url, false)
+    expect(span.context).toEqual({
+      http: {
+        method: 'GET',
+        url,
+        status_code: 200
+      },
+      destination: {
+        service: {
+          name: 'https://www.elastic.co',
+          resource: 'www.elastic.co:443',
+          type: 'external'
+        },
+        address: 'www.elastic.co',
+        port: '443'
+      }
+    })
+
+    url = 'http://[::1]'
+    span = createSpanWithData(url)
+    expect(span.context).toEqual({
+      http: {
+        method: 'GET',
+        url,
+        status_code: 200
+      },
+      destination: {
+        service: {
+          name: 'http://[::1]',
+          resource: '[::1]:80',
+          type: 'external'
+        },
+        address: '::1',
+        port: '80'
+      }
+    })
+
+    url = 'https://[::1]:80/'
```

Comment:
might be worth adding a comment, so it's clear this wasn't a typo :)

Reply:
sure, will do 👍